test/uk/gov/hmrc/cgtpropertydisposals/actors/TaxEnrolmentRetryManagerSpec.scala

```diff
@@ -0,0 +1,208 @@
+        "userId",
+        "someref",
+        Address.UkAddress("line1", None, None, None, "KO"),
+        "InProgress",
+        2,
+        0,
+        249237542,
+        LocalDateTime.now()
+      )
+
+      app ! RetryTaxEnrolmentRequest(taxEnrolmentRequest)
+      mongoProbe.receiveN(2)
+    }
+
+    "send a message to the connector actor if the elapsed time has exceeded the max elapsed time" in {
+      val mongoProbe     = TestProbe()
+      val connectorProbe = TestProbe()
+
+      val app = system.actorOf(
+        Props(
+          new TaxEnrolmentRetryManager(
+            (_: ActorRefFactory) => new FakeChildMaker(self, mongoProbe.ref).newChild(),
+            (_: ActorRefFactory) => new FakeChildMaker(self, connectorProbe.ref).newChild(),
+            mockTaxEnrolmentRepository,
+            mockTaxEnrolmentService,
+            system.scheduler
+          )
+        )
+      )
+
+      val taxEnrolmentRequest = TaxEnrolmentRequest(
+        "userId",
+        "someref",
+        Address.UkAddress("line1", None, None, None, "KO"),
+        "InProgress",
+        0,
+        0,
+        0,
+        LocalDateTime.of(2019, 12, 1, 12, 10)
+      )
+
+      app ! RetryTaxEnrolmentRequest(taxEnrolmentRequest)
+      // need to advance the time
+//      time.advance(FiniteDuration(10, TimeUnit.SECONDS))
+
+      connectorProbe.expectMsg(
+        RetryCallToTaxEnrolmentService(
+          TaxEnrolmentRequest(
+            "userId",
+            "someref",
+            UkAddress("line1", None, None, None, "KO"),
+            "InProgress",
+            1,
+            1,
+            0,
+            LocalDateTime.of(2019, 12, 1, 12, 10)
+          )
+        )
+      )
+    }
+
+    "must restart the mongo actor if it dies" in {
+      val mongoProbe     = TestProbe()
+      val connectorProbe = TestProbe()
+
+      system.actorOf(
+        Props(
+          new TaxEnrolmentRetryManager(
+            (_: ActorRefFactory) => new FakeChildMaker(self, mongoProbe.ref).newChild(),
+            (_: ActorRefFactory) => new FakeChildMaker(self, connectorProbe.ref).newChild(),
+            mockTaxEnrolmentRepository,
+            mockTaxEnrolmentService,
+            system.scheduler
+          )
+        )
+      )
+      mongoProbe.ref ! PoisonPill
+      expectMsg(NewChildCreated)
+    }
+
+    "must restart the connector actor if it dies" in {
+      val mongoProbe     = TestProbe()
+      val connectorProbe = TestProbe()
+
+      system.actorOf(
+        Props(
+          new TaxEnrolmentRetryManager(
+            (_: ActorRefFactory) => new FakeChildMaker(self, mongoProbe.ref).newChild(),
+            (_: ActorRefFactory) => new FakeChildMaker(self, connectorProbe.ref).newChild(),
+            mockTaxEnrolmentRepository,
+            mockTaxEnrolmentService,
+            system.scheduler
+          )
+        )
+      )
+      connectorProbe.ref ! PoisonPill
+      expectMsg(NewChildCreated)
```

Comment:
I'd include a test to see that the new child receives the messages you expect it to after it's created - you can make `NewChildCreated` include the actor ref of the new child to facilitate this